server/src/main/java/org/opensearch/action/search/SearchUtils.java

```diff
@@ -27,7 +26,7 @@ public SearchUtils() {}
     /**
      * Get connection lookup listener for list of clusters passed
      */
-    public static ActionListener<BiFunction<String, String, DiscoveryNode>> getConnectionLookupListener(
+    public static StepListener<BiFunction<String, String, DiscoveryNode>> getConnectionLookupListener(
```

Comment:
Lets keep this as ActionListener more generic and wrap a StepListener around it?